ssg/build_cpe.py

```diff
@@ -194,81 +199,73 @@ def to_xml_element(self):
 
         return cpe_test
 
-    def enrich_with_cpe_info(self, cpe_products):
+    def add_enriched_cpe_items(self, product_cpes):
         for arg in self.args:
-            arg.enrich_with_cpe_info(cpe_products)
-
-    def to_bash_conditional(self):
-        child_bash_conds = [
-            a.to_bash_conditional() for a in self.args
-            if a.to_bash_conditional() != '']
-
-        if not child_bash_conds:
-            return ""
-
-        cond = ""
-        if self.is_not():
-            cond += "! "
-            op = " "
-        cond += "( "
-        if self.is_or():
-            op = " || "
-        elif self.is_and():
-            op = " && "
-        cond += op.join(child_bash_conds)
-        cond += " )"
-        return cond
-
-    def to_ansible_conditional(self):
-        child_ansible_conds = [
-            a.to_ansible_conditional() for a in self.args
-            if a.to_ansible_conditional() != '']
-
-        if not child_ansible_conds:
-            return ""
-
-        cond = ""
-        if self.is_not():
-            cond += "not "
-            op = " "
-        cond += "( "
-        if self.is_or():
-            op = " or "
-        elif self.is_and():
-            op = " and "
-        cond += op.join(child_ansible_conds)
-        cond += " )"
-        return cond
-
-
-class CPEALFactRef (Symbol):
+            arg.add_enriched_cpe_items(product_cpes)
+
+    def get_remediation_conditional(self, language, product_cpes, conditionals_path, env_yaml):
```

Comment:
This function is difficult to read through, it needs naming tweaks and extractions as well.
Its scope is also unclear - is it supposed to construct a conditional for any language in which remediations can be written?